File: src/main/java/rars/exceptions/BreakpointException.java
package rars.exceptions;

/**
 * This exception is only used to trigger breakpoints for ebreak.
 * <p>
 * Its a bit of a hack, but it works and somewhat makes logical sense.
 */
public final class BreakpointException extends SimulationException {
    public static final BreakpointException BREAKPOINT_EXCEPTION = new BreakpointException();

    private BreakpointException() {
    }
}

File: src/main/java/rars/exceptions/BreakpointException.kt
package rars.exceptions

/**
 * This exception is only used to trigger breakpoints for ebreak.
 *
 *
 * It's a bit of a hack, but it works and somewhat makes logical sense.
 */
object BreakpointException : SimulationException(ExceptionReason.OTHER, null, 0) 
 

